rust/feature-flags/src/flags/flag_filters.rs

```diff
@@ -0,0 +1,253 @@
+use std::collections::HashMap;
+
+use serde_json::Value;
+
+use crate::flags::flag_models::{FlagFilters, FlagPropertyGroup};
+
+impl FlagFilters {
+    pub fn requires_db_properties(&self, overrides: &HashMap<String, Value>) -> bool {
+        self.aggregation_group_type_index.is_some()
+            || requires_db_properties(Some(&self.groups), overrides)
+            || requires_db_properties(self.super_groups.as_ref(), overrides)
+            || requires_db_properties(self.holdout_groups.as_ref(), overrides)
+    }
+
+    pub fn requires_cohort_filters(&self) -> bool {
+        requires_cohort_filters(Some(&self.groups))
+            || requires_cohort_filters(self.super_groups.as_ref())
+            || requires_cohort_filters(self.holdout_groups.as_ref())
+    }
+}
+
+fn requires_db_properties(
```

Comment:
maybe just me but having three different functions with the same name was a bit confusing at first.

Reply:
Oh, this one doesn't belong to a type. I'll definitely rename it.

Reply:
Done!